internal/build_defs.bzl

```diff
@@ -103,7 +103,7 @@ def _compile_action(ctx, inputs, outputs, tsconfig_file, node_opts, description
     # These deps are identified by the NodeModuleInfo provider.
     for d in ctx.attr.deps:
         if NodeModuleInfo in d:
-            action_inputs.extend(_filter_ts_inputs(d.files))
+            action_inputs.extend(_filter_ts_inputs(d.files.to_list()))
```

Comment:
`to_list()` has some performance issues and should be avoided ([see ref](https://docs.bazel.build/versions/master/skylark/performance.html#avoid-calling-depsetto_list)).

Reply:
I think this is subject to the performance problem either way. The question is whether `to_list` is done implicitly by passing the depset to an API that expects a list, or done explicitly. I think the bazel team is moving us to the latter.

Reply:
Yes, it was just to make the behavior more obvious. Bazel implicitly flattens depsets in some places (which is a performance problem). We want to forbid that. The simplest way forward is to make all flattening explicit in existing files.